aiohttp/connector.py

```diff
@@ -964,6 +975,15 @@ async def _wrap_create_connection(
             async with ceil_timeout(
                 timeout.sock_connect, ceil_threshold=timeout.ceil_threshold
             ):
+                if self._network_interface:
+                    *args, sock_hostname, sock_port = args  # type: ignore
+                    s = connect_socket_with_interface(
+                        self._network_interface,
+                        family=kwargs["family"],
+                        local_addr=self._local_addr,
+                    )
+                    await self._loop.sock_connect(s, (sock_hostname, sock_port))
+                    kwargs["sock"] = s
                 return await self._loop.create_connection(*args, **kwargs)  # type: ignore[return-value]  # noqa
```

Comment:
```suggestion
                conn_args: Sequence[Any]
                if self._network_interface:
                    *conn_args, sock_hostname, sock_port = args
                    s = connect_socket_with_interface(
                        self._network_interface,
                        family=kwargs["family"],
                        local_addr=self._local_addr,
                    )
                    await self._loop.sock_connect(s, (sock_hostname, sock_port))
                    kwargs["sock"] = s
                else:
                    conn_args = args
                return await self._loop.create_connection(*conn_args, **kwargs)  # type: ignore[return-value]  # noqa
```

Could be one possible approach. I'm not sure if there's an easy way to just declare the type of `args` itself, as the `tuple` part is inferred and not available.

Reply:
I think it's more simple to just pass `(args[-2], args[-1])` to sock_connect, no?

Reply:
Maybe use `conn_args: Sequence[Any] = args` before the if-block and let it rewrite this value conditionally. No else-block needed.

Reply:
> more simple to just pass `(args[-2], args[-1])` to sock_connect, no?

That does not contribute to the readability/maintainability of the code base. It shouldn't be only understood by a person adding changes in the first 5 minutes, but also 5 years from now. Things like `args[-2]` are confusing and don't have a named label to suggest what's the semantic meaning of the thing being passed. We could've made calls like `thing(x, y, z)` but tell be what this means instantly without having to inspect what `x`, `y` and `z` represent without reading another 100 lines of code that may have complex transformations too.
Naming pieces of data in code is beyond important — it allows humans to reason about the higher-level logic without an unnecessary level of mental overhead which increases the maintenance burden because of unreadable code.

Reply:
@Mostafa-Mahdi note that you can use GH `Commit suggestion` buttons to make sure you credit the co-authors of the patch. https://hynek.me/til/easier-crediting-contributors-github/ explains why this is important.